### ISSUE TITLE
operator crashes on crd vmagent.spec.remotewrite.tlsconfig

### ISSUE THREAD

**Issue by hamelg** [VictoriaMetrics/operator#215]
This simple crd triggers a SIGSEGV when the Field tlsConfig is present, whatever TLSConfig configuration parameters.
I use the version v0.12.2 from the OperatorHub.
It stops crashing when the tlsConfig section is absent.

```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMAgent
metadata:
  finalizers:
    - apps.victoriametrics.com/finalizer
  generation: 5
  name: example-vmagent
  namespace: w2-xxxx
spec:
  remoteWrite:
    - tlsConfig:
        insecureSkipVerify: true
      url: https://vm:vm@victoria.net.intra.xxxxxx.fr/api/v1/write
  replicaCount: 1
```

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x151409a]

goroutine 644 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x1763520, 0x2554af0)
        runtime/panic.go:969 +0x1b9
github.com/VictoriaMetrics/operator/controllers/factory.BuildRemoteWrites(0xc001d2a478, 0xc000abf710, 0xc000abf740, 0x1, 0x1, 0xc00519ea00)
        github.com/VictoriaMetrics/operator/controllers/factory/vmagent.go:812 +0x1cfa
github.com/VictoriaMetrics/operator/controllers/factory.makeSpecForVMAgent(0xc006316478, 0xc0005a8000, 0xc000abf710, 0xc000abf740, 0x1f, 0x9, 0x17)
        github.com/VictoriaMetrics/operator/controllers/factory/vmagent.go:208 +0x4a30
github.com/VictoriaMetrics/operator/controllers/factory.newDeployForVMAgent(0xc000e6f200, 0xc0005a8000, 0xc000abf710, 0xc000abf740, 0x0, 0x0, 0xc000fad630)
        github.com/VictoriaMetrics/operator/controllers/factory/vmagent.go:166 +0x165
github.com/VictoriaMetrics/operator/controllers/factory.CreateOrUpdateVMAgent(0x1b5a8c0, 0xc000f28a80, 0xc000e6f200, 0x1b6f6e0, 0xc0007ca3c0, 0xc0005a8000, 0x1, 0x0, 0x0, 0x0)
        github.com/VictoriaMetrics/operator/controllers/factory/vmagent.go:111 +0x5aa
github.com/VictoriaMetrics/operator/controllers.(*VMServiceScrapeReconciler).Reconcile(0xc0007ca3c0, 0x1b5a8c0, 0xc000f28a80, 0xc000f38f00, 0x2a, 0xc000f43200, 0x20, 0xc000f28a00, 0x0, 0x0, ...)
        github.com/VictoriaMetrics/operator/controllers/vmservicescrape_controller.go:90 +0x6a5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001759a0, 0x1b5a800, 0xc0007dbfc0, 0x17bd840, 0xc00087db20)
        sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:297 +0x317
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001759a0, 0x1b5a800, 0xc0007dbfc0, 0x0)
        sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:252 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x1b5a800, 0xc0007dbfc0)
        sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:215 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000057f50)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc006317f50, 0x1b21c60, 0xc000f28a20, 0xc0007dbf01, 0xc0000461e0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000057f50, 0x3b9aca00, 0x0, 0x3b9aca01, 0xc0000461e0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1b5a800, 0xc0007dbfc0, 0xc000fad550, 0x3b9aca00, 0x0, 0x100000000000001)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x1b5a800, 0xc0007dbfc0, 0xc000fad550, 0x3b9aca00)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:212 +0x45e
```
Any ideas ?

**Comment by f41gh7** [comment]
Hello, thanks for reporting, its a bug, will be fixed soon.

**Comment by f41gh7** [comment]
Fix was added to the 0.13.1 release. It should be available soon at operator hub.

**Comment by f41gh7** [comment]
New version released to operator hub, feel free to re-open an issue, if it doesn't help.
